lgc/elfLinker/GlueShader.h

```diff
@@ -76,6 +76,10 @@ class GlueShader {
   // Get the name of this glue shader.
   virtual llvm::StringRef getName() const = 0;
 
+  // Update the PAL metadata entries that require the glue code data and the
+  // pipeline state.
+  virtual void updatePalMetadata(PalMetadata *palMetadata) = 0;
```

Comment:
nit: Is it allowed to pass a null pointer? If not, consider passing the metadata by reference.

Reply:
That is a direction that the LLVM code base has been heading for some time, but we don't really do that consistently in LGC.

Reply:
Done.